src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.LoadDirective.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    public sealed partial class CSharpCompilation
+    {
+        private struct LoadDirective
+        {
+            public readonly SyntaxTree LoadedTree;
+            public readonly LoadDirectiveTriviaSyntax Directive;
+            public readonly ImmutableArray<Diagnostic> Diagnostics;
+
+            public LoadDirective(SyntaxTree loadedTree, LoadDirectiveTriviaSyntax directive, ImmutableArray<Diagnostic> diagnostics)
+            {
+                Debug.Assert(!diagnostics.IsDefault);
+                Debug.Assert((loadedTree == null) != diagnostics.IsEmpty);
```

Comment:
Yes, ```diagnostics``` should only contain errors.  I've added an Assert to that effect.  Also changed logic in other Assert.